ext/gvl_tracing_native_extension/gvl_tracing.c

```diff
@@ -177,9 +179,23 @@ static void set_native_thread_id(void) {
   thread_id = native_thread_id;
 }
 
+static const char* get_event_name(rb_event_flag_t event_id) {
+  switch (event_id) {
+    case RUBY_INTERNAL_THREAD_EVENT_READY:     return "wants_gvl";
+    case RUBY_INTERNAL_THREAD_EVENT_RESUMED:   return "running";
+    case RUBY_INTERNAL_THREAD_EVENT_SUSPENDED: return "waiting";
+    case RUBY_INTERNAL_THREAD_EVENT_STARTED:   return "started";
+    case RUBY_INTERNAL_THREAD_EVENT_EXITED:    return "died";
+    case RUBY_INTERNAL_EVENT_GC_ENTER:         return "gc";
+    // TODO: is it possible the thread wasn't running? Might need to save the last state.
```

Comment:
I don't think this can happen -- from what I've seen, Ruby always triggers GC in the thread that was running, since there's no background GC, so it's always something that the thread does that triggers GC.